Common/src/toolboxes/C1DInterpolation.cpp

```diff
@@ -0,0 +1,202 @@
+                                                        su2double Theta ,
+                                                        unsigned short nDim, 
+                                                        su2double *Coord, 
+                                                        unsigned short nVar_Turb,
+                                                        CConfig *config){
+    su2double size_columns=Inlet_Interpolated.size()+nDim;
+    vector<su2double> Inlet_Values (size_columns);
+    su2double unit_r, unit_Theta, unit_m, Alpha, Phi;
+
+    /*---For x,y,z,T,P columns---*/
+    for (int i=0;i<nDim+2;i++){
+        if (i<nDim)
+            Inlet_Values[i] = Coord[i];
+        else
+            Inlet_Values[i] = Inlet_Interpolated[nDim-2];
+    }
+
+    /*---For turbulence variables columns---*/
+    if (nVar_Turb == 1)
+        Inlet_Values[nDim+5] = Inlet_Values[5];
+    if (nVar_Turb == 2)
+        Inlet_Values[nDim+6] = Inlet_Values[6];
+
+    /*--- Correct for Interpolation Type now ---*/
+    switch(config->GetKindInletInterpolationType()){
+    case(VR_VTHETA):
+        unit_r = Inlet_Interpolated[nDim+2];
+        unit_Theta = Inlet_Interpolated[nDim+3];
+    break;
+    case(ALPHA_PHI):
+        Alpha = Inlet_Interpolated[nDim+2];
+        Phi = Inlet_Interpolated[nDim+3];
+        unit_m = sqrt(1/(1+pow(tan(Alpha),2)));
+        unit_Theta = tan(Alpha)*unit_m;
+        unit_r=unit_m*sin(Phi);
+    break;
+    }
+
+    Inlet_Values[nDim+2] = unit_r*cos(Theta) - unit_Theta*sin(Theta); //for ix
+    Inlet_Values[nDim+3] = unit_r*sin(Theta) + unit_Theta*cos(Theta); //for iy
+    Inlet_Values[nDim+4] = sqrt(1-pow(unit_r,2)- pow(unit_Theta,2));  //for iz
+
+    return Inlet_Values;
+
+}
+
+/*--- For printing interpolated data to a file ---*/
+void C1DInterpolation::PrintInletInterpolatedData(vector<su2double>& Inlet_Values, string Marker, unsigned long nVertex, unsigned short nDim){
+    ofstream myfile;
+    myfile.precision(16);
+    myfile.open("Interpolated_Data_"+Marker+".dat",ios_base::out);
+
+    if (myfile.is_open()){  
+    for (unsigned long iVertex = 0; iVertex < nVertex; iVertex++) {
+                for  (unsigned long iVar=0; iVar < Inlet_Values.size(); iVar++){
+                    myfile<<Inlet_Values[iVertex*Inlet_Values.size()+iVar]<<"\t";
+                }
+                myfile<<endl;
+            }
+    myfile.close();
+    }
+    else
+        cout<<"file cannot be opened"<<endl;
+}
+
+void C1DInterpolation::SetDataFromInletColumns(vector<su2double>& Inlet_Data, unsigned short nColumns, unsigned long nRows, unsigned short iCol, vector<su2double> &X, vector<su2double> &Data){
+    int n = Inlet_Data.size();
+    X.resize(n);
+    Data.resize(n);
+    
+    unsigned long index;
+    for (unsigned long iRow = 0; iRow < nRows; iRow++){
+    index = iRow*nColumns;
+        X[iRow]=Inlet_Data[index];
+        Data[iRow]=Inlet_Data[index+iCol];
+    }
+
+    SetSpline(X,Data);
+}
```

Comment:
Make this a method of the profileReader, for example `profileReader.GetDataForProfile(jMarker, iCol)`, then you can have just one constructor for the interpolation classes right?

Reply:
Yes that's correct. However, I am first thinking of resolving the segfaults before I go on here.